src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/SourceRepositoryExtensions.cs

```diff
@@ -87,6 +87,26 @@ public static Task<SearchResult<IPackageSearchMetadata>> SearchAsync(this Source
             return result;
         }
 
+        /// <summary>
+        /// Get the package metadata for the given identity
+        /// </summary>
+        public static async Task<IPackageSearchMetadata> GetPackageMetadataForIdentityAsync(this SourceRepository sourceRepository, PackageIdentity identity, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            var metadataResource = await sourceRepository.GetResourceAsync<PackageMetadataResource>(cancellationToken);
+
+            using (var sourceCacheContext = new SourceCacheContext())
+            {
+                // Update http source cache context MaxAge so that it can always go online to fetch latest version of packages.
+                sourceCacheContext.MaxAge = DateTimeOffset.UtcNow;
+
+                return await metadataResource?.GetMetadataAsync(identity, sourceCacheContext, Common.NullLogger.Instance, cancellationToken);
```

Comment:
If `metadataResource` is `null`, this statement will throw a `NullReferenceException`.  The null-conditional operator `?` will return `null`, but you can't `await` `null`.

Reply:
Good catch.